src/sys/ptrace.rs

```diff
@@ -1,66 +1,87 @@
+        PTRACE_SETOPTIONS  = libc::PTRACE_SETOPTIONS,
+        PTRACE_GETEVENTMSG = libc::PTRACE_GETEVENTMSG,
+        PTRACE_GETSIGINFO  = libc::PTRACE_GETSIGINFO,
+        PTRACE_SETSIGINFO  = libc::PTRACE_SETSIGINFO,
+        PTRACE_GETREGSET   = libc::PTRACE_GETREGSET,
+        PTRACE_SETREGSET   = libc::PTRACE_SETREGSET,
+        PTRACE_SEIZE       = libc::PTRACE_SEIZE,
+        PTRACE_INTERRUPT   = libc::PTRACE_INTERRUPT,
+        PTRACE_LISTEN      = libc::PTRACE_LISTEN,
+        PTRACE_PEEKSIGINFO = libc::PTRACE_PEEKSIGINFO,
+    }
 
-    pub const PTRACE_TRACEME:     PtraceRequest = 0;
-    pub const PTRACE_PEEKTEXT:    PtraceRequest = 1;
-    pub const PTRACE_PEEKDATA:    PtraceRequest = 2;
-    pub const PTRACE_PEEKUSER:    PtraceRequest = 3;
-    pub const PTRACE_POKETEXT:    PtraceRequest = 4;
-    pub const PTRACE_POKEDATA:    PtraceRequest = 5;
-    pub const PTRACE_POKEUSER:    PtraceRequest = 6;
-    pub const PTRACE_CONT:        PtraceRequest = 7;
-    pub const PTRACE_KILL:        PtraceRequest = 8;
-    pub const PTRACE_SINGLESTEP:  PtraceRequest = 9;
-    pub const PTRACE_GETREGS:     PtraceRequest = 12;
-    pub const PTRACE_SETREGS:     PtraceRequest = 13;
-    pub const PTRACE_GETFPREGS:   PtraceRequest = 14;
-    pub const PTRACE_SETFPREGS:   PtraceRequest = 15;
-    pub const PTRACE_ATTACH:      PtraceRequest = 16;
-    pub const PTRACE_DETACH:      PtraceRequest = 17;
-    pub const PTRACE_GETFPXREGS:  PtraceRequest = 18;
-    pub const PTRACE_SETFPXREGS:  PtraceRequest = 19;
-    pub const PTRACE_SYSCALL:     PtraceRequest = 24;
-    pub const PTRACE_SETOPTIONS:  PtraceRequest = 0x4200;
-    pub const PTRACE_GETEVENTMSG: PtraceRequest = 0x4201;
-    pub const PTRACE_GETSIGINFO:  PtraceRequest = 0x4202;
-    pub const PTRACE_SETSIGINFO:  PtraceRequest = 0x4203;
-    pub const PTRACE_GETREGSET:   PtraceRequest = 0x4204;
-    pub const PTRACE_SETREGSET:   PtraceRequest = 0x4205;
-    pub const PTRACE_SEIZE:       PtraceRequest = 0x4206;
-    pub const PTRACE_INTERRUPT:   PtraceRequest = 0x4207;
-    pub const PTRACE_LISTEN:      PtraceRequest = 0x4208;
-    pub const PTRACE_PEEKSIGINFO: PtraceRequest = 0x4209;
+    // These aren't currently in libc.
+    #[cfg(any(target_os = "linux", target_os = "android"))]
+    #[derive(Eq, PartialEq, Clone, Copy, Debug)]
+    #[repr(u32)]
+    pub enum PtraceEvent {
+        PTRACE_EVENT_FORK       = 1,
+        PTRACE_EVENT_VFORK      = 2,
+        PTRACE_EVENT_CLONE      = 3,
+        PTRACE_EVENT_EXEC       = 4,
+        PTRACE_EVENT_VFORK_DONE = 5,
+        PTRACE_EVENT_EXIT       = 6,
+        PTRACE_EVENT_SECCOMP    = 7,
+        PTRACE_EVENT_STOP       = 128,
+    }
 
-    pub type PtraceEvent = c_int;
+    #[cfg(any(target_os = "linux", target_os = "android"))]
+    impl PtraceEvent {
+        /// Creates a PtraceEvent from the extra bits of a wait status (status >> 16)
+        #[inline]
+        pub fn from_c_int(event: libc::c_uint) -> Option<PtraceEvent> {
```

Comment:
for `::nix::sys::signal::Signal` we used `Result<Signal>` as a return type for the corresponding function. I think this is better, because I don't think anyone would intentionally put a wrong `c_uint` into `from_c_int`, so the `None` return type is never expected, thus should probably be an error type.